core/src/main/java/org/pentaho/di/core/vfs/KettleVFS.java

```diff
@@ -132,71 +131,73 @@ public static FileObject getFileObject( String vfsFilename, FileSystemOptions fs
   // However, how are we going to verify this?
   //
   // We are going to see if the filename starts with one of the known protocols like file: zip: ram: smb: jar: etc.
-  // If not, we are going to assume it's a file, marking the relativeFilename flag as true, and it only changes if
+  // If not, we are going to assume it's a file, when no scheme found ( flag as null ), and it only changes if
   // a scheme is provided.
   //
   public static FileObject getFileObject( String vfsFilename, VariableSpace space, FileSystemOptions fsOptions )
     throws KettleFileException {
+
+    //  Protect the code below from invalid input.
+    if ( vfsFilename == null ) {
+      throw new KettleFileException( "Unexpected null VFS filename",
+        new NullPointerException( "KettleVFS.getFileObject() vfsFilename is NULL" ) );
+    }
+
     try {
-      fsOptionsForScheme = fsOptions;
       FileSystemManager fsManager = getInstance().getFileSystemManager();
       int timeOut = TIMEOUT_LIMIT;
-      boolean relativeFilename = true;
-      String[] initialSchemes = fsManager.getSchemes();
-
-      relativeFilename = checkForSchemeProvider( initialSchemes, relativeFilename, vfsFilename, space, fsOptionsForScheme );
+      String[] schemes = fsManager.getSchemes();
 
-      boolean hasScheme = vfsFilename != null && hasScheme( vfsFilename, PROVIDER_PATTERN_SCHEME );
+      String scheme = getScheme( schemes, vfsFilename );
```

Comment:
Aren't these redundant? Why not use the ones from the while cycle?

Reply:
Just to ensure we don't use always one thread cycle execution.